### PR TITLE
PushInteractor and APNS refactoring and simplication

### PR DESCRIPTION
**Pull request by malayli**

### CODE REVIEW

File: PusherMainView/PusherMainView/AuthTokenViewcontroller.swift [hunk 1]
```diff
@@ -10,6 +10,15 @@ final class AuthTokenViewcontroller: NSViewController {
     
     required init?(coder: NSCoder) {
         super.init(coder: coder)
+        #if DEBUG
+        print("\(self.className) init")
+        #endif
```
Comment (donnie-jp) [inline PusherMainView/PusherMainView/AuthTokenViewcontroller.swift]: remove?

Comment (malayli) [reply PusherMainView/PusherMainView/AuthTokenViewcontroller.swift]: I let it in purpose so we can see init and deinit calls in console.

File: PusherMainView/PusherMainView/PusherInteractor.swift [hunk 2]
```diff
@@ -140,12 +143,75 @@ extension PusherInteractor: PusherInteracting {
...
+    func unsubscribe(_ pusherInteractable: PusherInteractable) {
+        guard let index = subscribers.firstIndex(where: { pusherInteractable == $0 }) else {
+            return
+        }
+        subscribers.remove(at: index)
     }
     
-    func selectDevice(_ device: APNSServiceDevice) {
-        delegate?.didSelectDevicetoken(device.token, appBundleID: device.appID)
+    func dispatch(actionType: ActionType) {
```
Comment (donnie-jp) [inline PusherMainView/PusherMainView/PusherInteractor.swift]: In redux I think the state is changed in the reducer functions.

Comment (malayli) [reply PusherMainView/PusherMainView/PusherInteractor.swift]: Yes it is true. For this PR, I let state changes here.

**Pull request merged**